slick/src/main/scala/slick/lifted/Query.scala

```diff
@@ -155,6 +155,11 @@ sealed abstract class Query[+E, U, C[_]] extends QueryBase[C[U]] { self =>
     new WrappingQuery[(G, Query[P, U, Seq]), (T, Query[P, U, Seq]), C](group, key.zip(value))
   }
 
+  /** Specify part of a select statement for update and marked for row level locking */
+  def forUpdate: Query[E, U, C] = {
```

Comment:
Interesting. So it indeed makes sense to put it into the query language. It's a rather specific feature, with a SQL-specific name, but I can't think of a better design, either.